packages/backend-common/src/errors.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * A set of common business logic errors.
+ *
+ * The error handler middleware understands these and will translate them to
+ * well formed HTTP responses.
+ *
+ * While these are intentionally analogous to HTTP errors, they are not
+ * intended to be thrown by the request handling layer. In those places, please
+ * use e.g. the http-errors library.
+ */
+
+/**
+ * The request is malformed and cannot be processed.
+ */
+export class BadRequestError extends Error {
+  readonly cause?: Error;
+
+  constructor(message?: string, cause?: Error) {
+    super(message);
+    Object.setPrototypeOf(this, BadRequestError.prototype);
```

Comment:
Better do all of this stuff in a common CustomError base class instead.

+1 for keeping track of cause, I think we could even do something like ``super(`${message}, ${cause}`)`` when there is a cause.

Reply:
But does one not have to actually give the subclass type here? I am not sure if it CAN be delegated to a base class.

Reply:
Oh like `class X extends ErrorBase { constructor(...) { super(X); } }`?

Reply:
Fixed